File: client/tests/parallel.test.ts
import { test, expect } from '@playwright/test';

test.describe.configure({mode: 'parallel'});

test('Check Nirvana page contains Nirvana', async ({ page }) => {
  await page.goto('http://localhost:8080/');

  await page.getByText('Nirvana').locator('..').getByRole('button', { name: 'View', exact: true }).click();

  await expect(page).toHaveURL('http://localhost:8080/#/songs/1');
  await page.getByText('Nirvana').click();
});

test('Check Cypress Hill page contains Cypress Hill', async ({ page }) => {
    await page.goto('http://localhost:8080/');

    await page.getByText('Cypress Hill').locator('..').getByRole('button', { name: 'View', exact: true }).click();
    
    await expect(page).toHaveURL('http://localhost:8080/#/songs/4');
    await page.getByText('Cypress Hill').click();
});

File: client/tests/java/src/test/java/org/tests/ParallelTest.java
package org.tests;

import com.microsoft.playwright.*;
import com.microsoft.playwright.options.*;

import static com.microsoft.playwright.assertions.PlaywrightAssertions.assertThat;

import org.testng.annotations.Test;
import org.tests.utils.Constants;

public class ParallelTest {

    @Test
    public void nirvanaNameTest() {
        try (Playwright playwright = Playwright.create()) {
            Browser browser = playwright.chromium().launch(new BrowserType.LaunchOptions()
                    .setHeadless(false));
            BrowserContext context = browser.newContext();
            Page page = context.newPage();
            page.navigate(Constants.CLIENT_URL);
            page.getByText("Nirvana").locator("..")
                    .getByRole(AriaRole.LINK, new Locator.GetByRoleOptions().setName("VIEW")).click();

            assertThat(page.getByText("Nirvana")).isVisible();
        }
    }

    @Test
    public void cypressHillNameTest() {
        try (Playwright playwright = Playwright.create()) {
            Browser browser = playwright.chromium().launch(new BrowserType.LaunchOptions()
                    .setHeadless(false));
            BrowserContext context = browser.newContext();
            Page page = context.newPage();
            page.navigate(Constants.CLIENT_URL);
            page.getByText("Cypress Hill").locator("..")
                    .getByRole(AriaRole.LINK, new Locator.GetByRoleOptions().setName("VIEW")).click();

            assertThat(page.getByText("Cypress Hill")).isVisible();
        }
    }
}